src/globus_action_provider_tools/authentication.py

```diff
@@ -107,40 +111,34 @@ def _cached_introspect_call(self) -> GlobusHTTPResponse:
 
         return introspect_result
 
-    def introspect_token(self) -> GlobusHTTPResponse | None:
+    def introspect_token(self) -> GlobusHTTPResponse:
         introspect_result = self._cached_introspect_call()
-
-        # FIXME: convert this to an exception, rather than 'None'
-        # the exception could be raised in _verify_introspect_result
-        if not introspect_result["active"]:
-            return None
-
         self._verify_introspect_result(introspect_result)
         return introspect_result
 
     def _verify_introspect_result(self, introspect_result: GlobusHTTPResponse) -> None:
         """
         A helper which checks token introspect properties and raises exceptions on failure.
         """
+
+        if not introspect_result["active"]:
+            raise InactiveTokenError("The token is invalid or has been revoked")
```

Comment:
```suggestion
            raise InactiveTokenError("The token is invalid.")
```

Small personal preference here.
Possible reasons for being invalid that I know of:
- the string is not an Auth-issued token string -- maybe it doesn't even parse internally, etc.
- the token expired
- the token was revoked explicitly by the user
- the token was not explicitly revoked, but our team found it in public and revoked it (e.g. we find a refresh token in someone's GitHub repo)

So I'd just say "it's invalid" and leave the possibility of revocation as something covered by that statement.

Reply:
Done.